extension/bearertokenauthextension/bearertokenauth.go

```diff
@@ -171,6 +174,38 @@ func (b *BearerTokenAuth) RoundTripper(base http.RoundTripper) (http.RoundTrippe
 	}, nil
 }
 
+// Authenticate checks whether the given context contains valid auth data.
+func (b *BearerTokenAuth) Authenticate(ctx context.Context, headers map[string][]string) (context.Context, error) {
```

Comment:
Any reason not to use `http.Header` type?

Reply:
I tried it, but it seems it does not match the Authenticator interface.
```bash
2023/05/24 16:37:09 collector server run finished with error: cannot start pipelines: requested authenticator is not a server authenticator
```